pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -162,6 +162,9 @@ func (dswp *desiredStateOfWorldPopulator) findAndRemoveDeletedPods() {
 			}
 			// Skip non-memory backed volumes belonging to terminated pods
 			volume := volumeToMount.VolumeSpec.Volume
+			if volume == nil {
+				continue
+			}
 			if (volume.EmptyDir == nil || volume.EmptyDir.Medium != v1.StorageMediumMemory) &&
```

Comment:
We still need to check if the temporal storage is disk- or memory-backed.  The idea here was that we wanted to skip removing disk-backed temporal storage until the pod is removed from the API server, but remove memory-backed volumes as soon as the pod is in a terminal state.

I have a follow-on PR https://github.com/kubernetes/kubernetes/pull/37228 for 1.6 that removes this check entirely.